### PR TITLE
Device decommissioning.

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
Device decommissioning + test improvements.
@bboozzoo @mchalski @maciejmrowiec

### CODE REVIEW

File: resources/deployments/mongo/device_deployments.go [hunk 1]
```diff
@@ -368,3 +396,33 @@ func (d *DeviceDeploymentsStorage) AbortDeviceDeployments(deploymentId string) e
...
 }
+
+func (d *DeviceDeploymentsStorage) DecommissionDeviceDeployments(deviceId string) error {
+	if govalidator.IsNull(deviceId) {
+		return ErrStorageInvalidID
+	}
+
+	session := d.session.Copy()
+	defer session.Close()
+	selector := bson.M{"$and": []bson.M{
```
Comment (bboozzoo) [inline resources/deployments/mongo/device_deployments.go]: Can you indent this block of code?

File: resources/deployments/mongo/device_deployments.go [hunk 2]
```diff
@@ -368,3 +396,33 @@ func (d *DeviceDeploymentsStorage) AbortDeviceDeployments(deploymentId string) e
...
+		bson.M{StorageKeyDeviceDeploymentDeviceId: deviceId},
+		bson.M{
+			"status": bson.M{
```
Comment (bboozzoo) [inline resources/deployments/mongo/device_deployments.go]: use `StorageKeyDeviceDeploymentStatus`

File: resources/deployments/mongo/device_deployments.go [hunk 3]
```diff
@@ -368,3 +396,33 @@ func (d *DeviceDeploymentsStorage) AbortDeviceDeployments(deploymentId string) e
...
+				"$in": deployments.ActiveDeploymentStatuses(),
+			}},
+	}}
+
+	update := bson.M{
+		"$set": bson.M{
+			"status": deployments.DeviceDeploymentStatusDecommissioned,
```
Comment (bboozzoo) [inline resources/deployments/mongo/device_deployments.go]: use `StorageKeyDeviceDeploymentStatus`

File: resources/deployments/mongo/device_deployments.go (hunk 1 above)
Comment (kjaskiewiczz) [inline resources/deployments/mongo/device_deployments.go]: done

File: resources/deployments/mongo/device_deployments.go (hunk 2 above)
Comment (kjaskiewiczz) [inline resources/deployments/mongo/device_deployments.go]: done

File: resources/deployments/mongo/device_deployments.go (hunk 3 above)
Comment (kjaskiewiczz) [inline resources/deployments/mongo/device_deployments.go]: done

**Pull request merged**